internal/host/dhcpallocatecmd.go

```diff
@@ -0,0 +1,84 @@
+package host
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/network"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/models"
+)
+
+type dhcpAllocateCmd struct {
+	baseCmd
+	dhcpAllocateImage string
+	db                *gorm.DB
+}
+
+func NewDhcpAllocateCmd(log logrus.FieldLogger, dhcpAllocateImage string, db *gorm.DB) *dhcpAllocateCmd {
+	return &dhcpAllocateCmd{
+		baseCmd:           baseCmd{log: log},
+		dhcpAllocateImage: dhcpAllocateImage,
+		db:                db,
+	}
+}
+
+func asMAC(macStr string) *strfmt.MAC {
+	mac := strfmt.MAC(macStr)
+	return &mac
+}
+
+func (f *dhcpAllocateCmd) prepareParam(host *models.Host) (string, bool, error) {
```

Comment:
Seems `bool` value (`false`/`true`) is equal to check if a `string` was returned.
So you can run `if param != ""`